x-pack/solutions/security/plugins/security_solution/public/data_view_manager/hooks/use_browser_fields.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import type { BrowserFields } from '@kbn/timelines-plugin/common';
+import { type DataViewManagerScopeName } from '../constants';
+import { useDataView } from './use_data_view';
+import { getDataViewStateFromIndexFields } from '../../common/containers/source/use_data_view';
+
+export const useBrowserFields = (scope: DataViewManagerScopeName): BrowserFields => {
+  const { dataView } = useDataView(scope);
+
+  return useMemo(() => {
+    if (!dataView) {
+      return {};
+    }
+
+    const { browserFields } = getDataViewStateFromIndexFields(
```

Comment:
👍🏾 , I think we should update this function to only return browser fields, and re-name it `createBrowserFieldsFromDataViewFields` or `transformDataViewFieldsToBrowserFields` or something and move it into it's own file as I think we now have multiple `use_data_view` files in security solution. That can be in this PR or a follow up one